### ISSUE TITLE
Reverse Engineer - Ability to have seperate folders for DataContext and Models in same project

### ISSUE THREAD

**Issue by Moorepheus** [ErikEJ/EFCorePowerTools#197]
Would be nice to have separate inputs for the Data Context and Model path.

**Comment by Moorepheus** [comment]
Would be nice to have separate entries for the DataContext and Model path.

**Comment by ErikEJ** [comment]
You can run the code gen twice, and then specify the folder/project for each run (and select to only generate Models or DbContext)

**Comment by ErikEJ** [comment]
Ping? Are you unblocked?

**Comment by Moorepheus** [comment]
Sorry, was out of the office. Yes, I understand you can run it twice, but that is an extra step that can be avoided.  Also, too many mistakes can be done having to run it twice.  Thanks for the help anyway.

**Comment by ErikEJ** [comment]
Reopening to enable upvotes

**Comment by tyrinj1120** [comment]
This would be a really nice feature. I was actually looking around the ui for it. Great tools Erik.

**Comment by ErikEJ** [comment]
@tyrinj1120 You can do this today by running the process twice, did you try that?

**Comment by Moorepheus** [comment]
> @tyrinj1120 You can do this today by running the process twice, did you try that?

Yes, but this leaves the door wide open for mistakes.  While I can do this all in one command line, this tool saves on having to type it out what I want to do, and if the ability was to have this feature in this product would also make it more efficient.

**Comment by ErikEJ** [comment]
@Moorepheus So you would like to text boxes on the form, by default same path in both?

**Comment by Moorepheus** [comment]
Yes, one text box for the Data Context Path and one text box for the Models path. They could be the same or different, depending on what the user wants to do. In our case, they are different. , This would make it more flexible then having to run it twice just because we want different paths for the Data Context and the Models.

**Comment by ErikEJ** [comment]
>> , by default same path in both?

Thansk, this was a question about the initial values!

**Comment by Moorepheus** [comment]
I think the initial values could be the same, I don't see how that would be an issue.

**Comment by jim-paton** [comment]
Is there any news on this? Awaiting this feature is currently blocking our adoption of the extension.

**Comment by ErikEJ** [comment]
Why are you blocked? https://github.com/ErikEJ/EFCorePowerTools/issues/197#issuecomment-477608575

**Comment by jim-paton** [comment]
As per some of the previous comments, we currently use this feature of the command line `dbcontext scaffold` command - we generate our context class into the root of the project and the model classes into a Models sub-folder. Our current workflow has a number of issues that the extension would solve, but without this feature we would simply be swapping one set of workarounds for another.

**Comment by ErikEJ** [comment]
I will look into this after reverse engineering for EF Core 3.0 has been done (including reverse engineering of views)

**Comment by ErikEJ** [comment]
@jim-paton Could you show me a sample command to put the DbContext in the root folder, and the model classes in a Models folder?

**Comment by ErikEJ** [comment]
Work on this is now in progress

https://github.com/aspnet/EntityFrameworkCore/blob/24b9aa1d2e14fe2e737255ede9b2a7a623fcf2af/src/EFCore.Design/Design/Internal/DatabaseOperations.cs#L64 

https://github.com/aspnet/EntityFrameworkCore/issues/18238

**Comment by ErikEJ** [comment]
TODO:

- [ ] Add new namespace logic to EF Core 2 reveng class
- [ ] Enable 3 new properties in view model and XAML binding
- [ ] Add 3 new properties to ReverseEngineerOptions
- [ ] Map 3 new property values to ReverseEngineerCommandOptions
- [ ] Testing/verification

**Comment by jespersh** [comment]
@ErikEJ would that allow me to have DbContext in one project and Entities in another?

**Comment by ErikEJ** [comment]
No, this is not what this feature is about, but you can already do that by specifying what to generate

**Comment by jespersh** [comment]
@ErikEJ alright. Just the "total control of namespaces and folder paths" that had me excited

**Comment by ErikEJ** [comment]
Might be possible - I will investigate

**Comment by ErikEJ** [comment]
@jespersh Full control: https://github.com/ErikEJ/EFCorePowerTools/commit/acbfd40d36e6f088dbe4fbe18822460b711e151a
